lib/hexstring.ml

```diff
@@ -1,17 +1,35 @@
 (* encoding bytearray -> hexstring *)
 
 let encode (bytearray : bytes) : string =
-  let rec encode_rec fold bytearray =
-    let length = Bytes.length bytearray in
-    if length = 0 then fold
+  let dummy_char = '_' in
+  let start_of_digit_0_in_ascii_table = 0x30 in
+  let start_of_lower_case_a_in_ascii_table = 0x61 in
+  let hex_digit_of_int (x : int) : char =
+    assert (x >= 0);
+    assert (x < 16);
+    char_of_int (
+      if x < 10 then
+        x + start_of_digit_0_in_ascii_table
+      else
+        x + start_of_lower_case_a_in_ascii_table
+    )
+  in
+  let rec aux bytearray len cur_pos buf =
```

Comment:
what does `aux` stand for here?

Reply:
Just `auxiliary` - seems to be the standard name for the generic internal loop `aux` in the ocaml community if I'm not mistaken.

Idm renaming it, but if there is only one loop in there then "eh".